sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSourceTransformWithStateSuite.scala

```diff
@@ -997,4 +997,85 @@ class StateDataSourceTransformWithStateSuite extends StateStoreMetricsTest
       }
     }
   }
+
+  /**
+   * Note that we cannot use the golden files approach for transformWithState. The new schema
+   * format keeps track of the schema file path as an absolute path which cannot be used with
```

Comment:
We may need to think through the idea of using "absolute path" - if the path is under checkpoint location, having the path as absolute path is unnecessarily limiting the flexibility. Ideally saying we should use relative path, unless we figure out the case where we see difference between base path + relative path vs absolute path.

But let's have another ticket to track this. Could you please file an JIRA issue for this? We could give a try with updating this test to use golden file again in that JIRA ticket.

Reply:
Sure - I will file a separate ticket for this